### DISCUSSION TITLE
webgl support?

### DISCUSSION THREAD

**Discussion by stylehuan** [bitbof/klecks#48] [Q&A] [answered]
How do I start WebGL

**Comment by bitbof** [comment]
This site has some infos https://get.webgl.org/

**Comment by stylehuan** [comment]
Sorry, what I meant to say is how to render in Klecks using WebGL

**Comment by bitbof** [comment, marked as answer]
Klecks does some WebGL drawing through glfx.js. You can see its usage in filters, e.g. `src/app/script/klecks/filters/filter-to-alpha.ts`

``` ts
let glCanvas = getSharedFx(); // get the glfx canvas (uses webgl)
let texture = glCanvas.texture(context.canvas); // turn current layer into a webgl texture
glCanvas.draw(texture).toAlpha(sourceId === 'inverted-luminance', selectedRgbaObj).update(); // apply toAlpha operation
// draw result back on layer
context.clearRect(0, 0, context.canvas.width, context.canvas.height);
context.drawImage(glCanvas, 0, 0);
texture.destroy(); // cleanup
```

Glfx code lives in `src/app/script-vendor/glfx.ts`.

Atm, everything outside of filters is done via CanvasRenderingContext2D.
